frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -154,12 +158,86 @@ private IGitRepositorySelectionProvider getGitRepositorySelectionProvider() {
 
   private void subscribeToGitRepositoryFilesChanges() {
     Topic<GitRepositoryChangeListener> topic = GitRepository.GIT_REPO_CHANGE;
-    GitRepositoryChangeListener listener = repository -> queueRepositoryUpdateAndModelRefresh();
+    GitRepositoryChangeListener listener = repository -> {
+      trackCurrentBranchChange(repository);
+      queueRepositoryUpdateAndModelRefresh();
+    };
+
     val messageBusConnection = project.getMessageBus().connect();
     messageBusConnection.subscribe(topic, listener);
     Disposer.register(project, messageBusConnection);
   }
 
+  private Notification getNotificationAboutNonManagedBranch(String branchName) {
+    val notification = VcsNotifier.STANDARD_NOTIFICATION.createNotification(
+        getString("action.GitMachete.EnhancedGraphTable.non-managed-branch-notification.text").format(branchName),
+        NotificationType.INFORMATION);
+    notification.addAction(
+        NotificationAction
+            .createSimple(getString("action.GitMachete.EnhancedGraphTable.non-managed-branch-notification.action"), () -> {
+              notification.expire();
+              val gitRepositorySelectionProvider = getGitRepositorySelectionProvider();
+              val gitRepository = gitRepositorySelectionProvider.getSelectedGitRepository();
+              if (gitRepository == null) {
+                LOG.warn("Selected repository is null");
+                return;
+              }
+
+              // A bit of a shortcut: we're accessing filesystem even though we're on the UI thread here;
+              // this shouldn't ever be a heavyweight operation, however.
+              Path macheteFilePath = gitRepository.getMacheteFilePath();
+              boolean isMacheteFilePresent = Files.isRegularFile(macheteFilePath);
+
+              if (isMacheteFilePresent) {
+                new SlideInUnmanagedBranch(
```

Comment:
I'd rather extract `SlideInUnmanagedBranchBackgroundable` and `.queue()` here from the calling class; the existing `SlideInUnmanagedBranch` seems to correspond to a special kind of _branch_ rather than a _task_